lib/model/model.go

```diff
@@ -1054,9 +1150,40 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 		if !ok {
 			// Shouldn't happen because !cfg.Paused, but might happen
 			// if the folder is about to be unpaused, but not yet.
+			l.Debugln("ccH: no fset", folder.ID)
 			continue
 		}
 
+		ccDeviceRemote, hasCCDeviceRemote := ccDevicesRemote[folder.ID]
+		ccDeviceLocal, hasCCDeviceLocal := ccDevicesLocal[folder.ID]
+
+		if err := m.ccCheckEncryption(cfg, folderDevice, ccDeviceRemote, ccDeviceLocal, hasCCDeviceRemote, hasCCDeviceLocal, deviceCfg.Untrusted); err != nil {
+			sameError := false
+			if devs, ok := m.folderEncryptionFailures[folder.ID]; ok {
+				sameError = devs[deviceID] == err
+			} else {
+				m.folderEncryptionFailures[folder.ID] = make(map[protocol.DeviceID]error)
+			}
+			m.folderEncryptionFailures[folder.ID][deviceID] = err
```

Comment:
Can this live in the folder rather than the model somehow?

Reply:
These are connection/cluster config level errors. And folders do not handle that let allow know anything about it. Changing that would be entirely out of scope of this PR.